xdsl/irdl/irdl.py

```diff
@@ -81,12 +81,32 @@ class IRDLAnnotations(Enum):
 #
 
 
+@dataclass
+class ConstraintContext:
+    """
+    Context for constraint variables.
```

Comment:
```suggestion
    Contains the assignment of constraint variables.
```
Otherwise the comment was not really helping much I think ;)